### PR TITLE
clj-kondo: init at 2019.07.31-alpha

### PR DESCRIPTION
**Pull request by jlesquembre**
###### Motivation for this change
Add [clj-kondo](https://github.com/borkdude/clj-kondo), a linter for clojure

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/clj-kondo/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  meta = with lib; {
+    description = "A linter for Clojure code that sparks joy.";
+    homepage = https://github.com/borkdude/clj-kondo;
+    license = licenses.epl10;
+    platforms = platforms.unix;
+    maintainers = with maintainers; [ jlesquembre ];
+  };
+
+
+  buildInputs = [ graalvm8 ]; # graalvm8 (as well as any other jvm) has a hook which form $CLASSPATH containing share/java/*.jar of all dependendencies
```
Comment (volth) [inline pkgs/development/tools/clj-kondo/default.nix]: I think the comment is not relevant here, because `$CLASSPATH` is not used

**Comment by jlesquembre** [comment]
No idea why I'm getting that error from grahamcofborg, locally works

File: pkgs/development/tools/clj-kondo/default.nix (hunk 1 above)
Comment (jlesquembre) [reply pkgs/development/tools/clj-kondo/default.nix]: You are right, I'll remove it

File: pkgs/development/tools/clj-kondo/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  src = fetchurl {
+    url = "https://github.com/borkdude/${pname}/releases/download/v${version}/${pname}-${version}-standalone.jar";
+    sha256 = "03ipl7br9pgx2hdbiaxv9ip0ibafkyzkc8qlx8xyi528bcfi54bf";
+  };
+  sourceRoot = "./";
+  unpackCmd = ''
+    ttfName=$(basename $(stripHash $curSrc))
+    cp $curSrc ./$ttfName
+  '';
```
Comment (volth) [inline pkgs/development/tools/clj-kondo/default.nix]:
```diff
 - sourceRoot = "./";
-  unpackCmd = ''
-    ttfName=$(basename $(stripHash $curSrc))
-    cp $curSrc ./$ttfName
-  '';
+  dontUnpack = true;
```

File: pkgs/development/tools/clj-kondo/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    license = licenses.epl10;
+    platforms = platforms.unix;
+    maintainers = with maintainers; [ jlesquembre ];
+  };
+
+
+  buildInputs = [ graalvm8 ]; # graalvm8 (as well as any other jvm) has a hook which form $CLASSPATH containing share/java/*.jar of all dependendencies
+  buildPhase = ''
+    native-image  \
+      -jar clj-kondo-${version}-standalone.jar \
```
Comment (volth) [inline pkgs/development/tools/clj-kondo/default.nix]:
```diff
- -jar clj-kondo-${version}-standalone.jar \
+ -jar $src
```

**Comment by volth** [comment]
> No idea why I'm getting that error from grahamcofborg, locally works

I suspect it is evaluation on Darwin.
Try to
```diff
- platforms = platforms.unix;
+ platforms = graalvm8.meta.platforms;
```

**Comment by jlesquembre** [comment]
@volth you are right, it was the evaluation on Darwin
Thanks a lot for your help with this PR, now looks a lot better :)

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build clj-kondo

**Review by worldofpeace — APPROVED** [review]
I can see the help menus, builds for me :+1:

**Comment by worldofpeace** [comment]
@jlesquembre Thanks for your patience.

**Pull request merged**